cli/compose/loader/loader.go

```diff
@@ -808,28 +808,58 @@ var transformStringList TransformerFunc = func(data any) (any, error) {
 	}
 }
 
-func transformMappingOrListFunc(sep string, allowNil bool) TransformerFunc {
-	return func(data any) (any, error) {
-		return transformMappingOrList(data, sep, allowNil), nil
-	}
-}
+var transformHostsList TransformerFunc = func(data any) (any, error) {
+	// Transform pairs of host and IP addr that may be represented as a map, or a list
+	// of '=' or ':' separated strings, into a list of ':' separated strings.
+	l := transformListOrMapping(data, ":", false, []string{"=", ":"})
 
-func transformListOrMappingFunc(sep string, allowNil bool) TransformerFunc {
-	return func(data any) (any, error) {
-		return transformListOrMapping(data, sep, allowNil), nil
+	// Remove brackets from IP addresses if present (for example "[::1]" -> "::1").
+	hl := l.([]string)
+	result := make([]string, 0, len(hl))
+	for _, hip := range hl {
+		host, ip, _ := strings.Cut(hip, ":")
+		if len(ip) > 2 && ip[0] == '[' && ip[len(ip)-1] == ']' {
+			ip = ip[1 : len(ip)-1]
+		}
+		result = append(result, fmt.Sprintf("%s:%s", host, ip))
 	}
+	return result, nil
 }
 
-func transformListOrMapping(listOrMapping any, sep string, allowNil bool) any {
+func transformListOrMapping(listOrMapping any, sep string, allowNil bool, allowSeps []string) any {
```

Comment:
Could it make sense to change the return type here to `[]string` instead of `any` now that these are the only options?

Reply:
Yes, might as well - done.